src/quark-core/src/QuarkNonceManager.sol

```diff
@@ -58,7 +48,8 @@ contract QuarkNonceManager {
             revert InvalidSubmissionToken(msg.sender, nonce, submissionToken);
         }
 
-        nonceSubmissions[msg.sender][nonce] = submissionToken;
+        // Note: even with a valid submission token, we always set non-replayables to exhausted (e.g. for cancelations)
+        submissions[msg.sender][nonce] = isReplayable ? submissionToken : EXHAUSTED;
```

Comment:
Makes sense, though it does mean that replayables can only be cancelled if the submitter has the next valid submission token in the hash chain. Seems fine, but it is a bit scary.

Reply:
will address in follow-up